src/internal/controllers/users_controller.go

```diff
@@ -3,64 +3,67 @@ package controllers
 import (
 	"birdai/src/internal/models"
 	"birdai/src/internal/repositories"
+	"birdai/src/internal/utils"
+
 	"go.mongodb.org/mongo-driver/bson"
 )
 
-func (c *Controller) CGetUserById(authId string) (*models.User, error) {
-	user, err := c.auth.CheckUser(authId)
-	if err != nil {
-		return &models.User{}, err
+func (c *Controller) CGetUserById(id string) (models.Response) {
+	coll := c.db.GetCollection(repositories.UserColl)
+	filter := bson.M{"_id": id}
+	response := coll.FindOne(filter)
+
+	if utils.IsTypeError(response) {
+		return response
 	}
-	return user.(*models.User), err
-}
 
-func (c *Controller) CLogin(user *models.User) (string, error) {
-	authUser, err := c.auth.LoginUser(user)
-	if err != nil {
-		return "", err
+	if utils.IsTypeUser(response) && response.Data.(models.User).Active == false {
+		return utils.ErrorDeleted("User collection")
 	}
-	return authUser, nil
+
+	return response
 }
 
-func (c *Controller) CListUsers(authId string) ([]*models.User, error) {
-	_, err := c.auth.CheckUser(authId)
-	if err != nil {
-		return nil, err
+func (c *Controller) CLoginUser(user *models.User) (models.Response) {
+	coll := c.db.GetCollection(repositories.UserColl)
+	response := coll.CreateOne(user)
+	if utils.IsTypeError(response) {
+		return response
 	}
+	return response
+}
+
+func (c *Controller) CListUsers() (models.Response) {
 	coll := c.db.GetCollection(repositories.UserColl)
-	inter, err := coll.FindAll()
-	users := inter.([]*models.User)
+	response := coll.FindAll()
+	users := []*models.User{}
 
-	return users, err
-}
+	if utils.IsTypeError(response) {
+		return response
+	}
 
-func (c *Controller) CDeleteUser(id, authId string) (*models.User, error) {
-	_, err := c.auth.CheckUser(authId)
-	if err != nil {
-		return nil, err
+	for _, usersObject := range response.Data.([]*models.User) {
+		users = append(users, usersObject)
 	}
+
+	return utils.Response(users)
+}
+
+func (c *Controller) CDeleteUser(id string) (models.Response) {
```

Comment:
Lär kolla om personens authId stämmer med mongoDB IDt
```suggestion
func (c *Controller) CDeleteUser(id, authId string) (models.Response) {
        response := c.auth.CheckUser(authId)
```